Work is done - PR to be controlled

Estelle-Mchd opened the pull request.

Estelle-Mchd committed:
Work is done - PR to be controlled

MrLootman commented:
Conflit avec `main` à résoudre.

MrLootman commented on client/src/components/UserSpace/UserSpace.tsx:
```diff
@@ -1,20 +1,26 @@
+      <div className="logo-savorsafari">
+        <img src="/public/img/SavorLogo.png" alt="Logo SavorSafari" />
+      </div>
+      <section className="hello-user">
```
Comme dit tout à l'heure, les noms de classe partout sont contre-productifs.

Dans ton CSS, si t u veux styliser ta section, tu peux la sélectionner comme suit :

```css
.userspace-element section:first-of-type {
}
```

Estelle-Mchd replied:
j'ai modifié :)

MrLootman commented on client/src/components/UserSpace/UserSpace.tsx:
```diff
@@ -1,20 +1,26 @@
+      </section>
+      <section className="list-of-recipe-consulted">
+        <ConsultedRecipe />
+      </section>
```
Si tu veux styliser celle-ci :

```css
.userspace-element section:nth-child(2) {

}
```

Estelle-Mchd replied:
j'ai modifié :)

MrLootman commented on client/src/components/UserSpace/ConsultedRecipe.css:
```diff
@@ -0,0 +1,38 @@
+    height: auto;
+    margin: 1rem;
+    padding: 1rem;
+  }
```
Tantôt des classes, tantôt des id... La vraie question, c'est pourquoi ?

Estelle-Mchd replied:
Pour montrer que je savais appelé les deux types :)

MrLootman reviewed (changes requested):
**Commentaire général :** Il faut impérativement lever le pied sur les className et les id, autant que possible.
Les id sont à proscrire autant que possible, car ils ajoutent inutilement de la priorité sur des éléments.
Les className sont à diminuer, car tu as tendance à en mettre partout, alors que des pseudo classes te permettent d'accéder aux éléments du DOM sans eux.
Je te laisse consulter les exemples pour t'inspirer à ce propos.

MrLootman commented on client/src/components/UserSpace/ConsultedRecipe.tsx:
```diff
@@ -1,17 +1,56 @@
+interface Recipe {
+  strMeal: string;
+  idMeal: string;
+}
```
Cette interface doit être isolée dans un fichier à part (`Recipe.ts`, par exemple).
Le dossier qui contient les types se situe dans le répertoire `src`.

Estelle-Mchd replied:
j'ai modifié :)

MrLootman commented on client/src/components/UserSpace/ConsultedRecipe.tsx:
```diff
@@ -1,17 +1,56 @@
+    };
+
+    getRecipes(4);
+  }, []);
```
Quel est le sens de toute cette logique ?

Estelle-Mchd replied:
cela me permet d'appeler 4fois pour avoir mes 4recettes aléatoires s'affichant dans mon éléments, si je met 3 j'aurais seulement 3 recettes etc

MrLootman replied:
Le problème ici, c'est l'intelligibilité de ton code.

Je te propose une refactorisation comme suit :

Tu crées une fonction qui génère quatre recettes aléatoires qui, une fois réceptionnées, mettent à jour la variable d'état `recipes`.

Dans ton `useEffect`, tu exécutes cette fonction une fois au montage du composant.

Ainsi, tu gagnes en lisibilité, en clarté, et tu utilises des concepts que tu as appris avec JS, tel que la boucle for. A noter que la présence de la boucle for démontre que tu vas requêter 4 fois l'API. C'est auto explicite.

```js
  const [recipes, setRecipes] = useState<ConsultedRecipe[]>([]);

  function getRecipes() {
    const fetchedRecipes: ConsultedRecipes[] = [];
    
    for (let i = 0; i < 4; i++) {
      fetch("https://www.themealdb.com/api/json/v1/1/random.php")
        .then((response) => response.json())
        .then((data) => fetchedRecipes.push(data.meals[0]))
    }
    
    setRecipes(fetchedRecipes)
  }
  
    useEffect(() => {
      getRecipes();
  }, []);
```

Estelle-Mchd committed:
Conflicts resolved

Estelle-Mchd committed:
ok
Merge branch 'dev' of github.com:wildcodeschool-2025-03/JS-WestDevs-P2-SavorSafari into feature/UsersRecipe

Estelle-Mchd committed:
Conflicts resolved bis

Estelle-Mchd committed:
PR resolved - className

MrLootman reviewed (approved):
👍
